code/menuui/mainhallmenu.cpp

```diff
@@ -917,6 +918,12 @@ void main_hall_do(float frametime)
 
 				// clicked on the tech room region
 				case TECH_ROOM_REGION:
+					// Check if Techroom should auto-switch to singleplayer mode --wookieejedi
+					if (Techroom_autoswitches_to_singleplayer) {
+						Player->flags &= ~PLAYER_FLAGS_IS_MULTI;
+						Game_mode = GM_NORMAL;
+					}
+
```

Comment:
I really think the actual logic for this should be in Freespace.cpp during the game state change.  And we need to keep track of whether or not we were in Multiplayer before, and here's why:

If we were in multi mode where the mainhall has only a ready room region without a dedicated multiplayer region, then switching to the tech room will force it to single player, and the user will have to go to the barracks to switch it back to multiplayer so that the ready room region will take them to the PXO or game list screen.

Reply:
Ah yeah that's a good point. I reckon moving all the logic into that same space would be useful? IE taking out the existing little chunks here and moving all the mode switch logic to the one section during the game state change.

Reply:
Heh as it turns out, going through all that checking made me realize that I can do all of this via existing scripting, so no need to clutter up state change code with a new PR.